src/mp-instance.ts

```diff
@@ -185,6 +189,8 @@ export default function mParticleInstance(this: IMParticleWebSDKInstance, instan
 
                     completeSDKInitialization(apiKey, mergedConfig, this);
                 });
+
+                self._IsSelfHosted = true;
```

Comment:
Just incase anything errors out, I'd raise this to the top of the if statement to line 177 to confirm it is always set.